useIsShake.js

```diff
@@ -69,15 +50,13 @@ export function useIsShake() {
   };
 
   useEffect(() => {
-    _subscribe();
-
-    const onLoad = setTimeout(() => {
-      isToggleReady = true;
-      clearTimeout(onLoad);
-    }, SVG_LOAD_DURATION);
-
+    // invocation of _subscribe when isToggleReady === true
+    if (isShakeReady) {
+      _subscribe();
+      // isShake Accelerometer listener is removed when !isToggleReady
+    } else _unsubscribe();
     return () => _unsubscribe();
-  }, []);
+  }, [isShakeReady]);
```

Comment:
This useEffect logic is adding and removing the Accelerometer listener dependent on the value of `isShakeReady` for debouncing triggering animation toggling.

Reply:
HOT